src/pages/index.js

```diff
@@ -16,79 +22,247 @@ const cardListEl = document.querySelector(".cards__list");
-    items: initialCards,
-    renderer: renderCard,
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "75623fcf-ddee-44c9-84c0-daef735bad5c",
+    "Content-Type": "application/json",
   },
-  ".cards__list"
-);
+});
+
+//api on page loading
 
-section.renderItems();
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    section = new Section(
+      {
+        items: initialCards,
+        renderer: renderCard,
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getUserInfo()
+  .then((userInfoData) => {
+    userInfo.setUserInfo(userInfoData);
+    userInfo.setUserProfileImage(userInfoData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+//instantiating other classes
 
 const popupWithImage = new PopupWithImage("#card-image");
-popupWithImage.setEventListeners();
 
 const newCardModal = new PopupWithForm(
   "#add-card-modal",
   handleAddCardFormSubmit
 );
-newCardModal.setEventListeners();
 
 const editCardModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
+
+const deleteCardModal = new DeleteConfirmation(
+  "#delete-card-confirmation",
+  deleteCardForm,
+  handleDeleteCard
+);
+
+const profilePictureModal = new PopupWithForm(
+  "#change-profile-picture-modal",
+  handleChangeProfilePictureSubmit
+);
+
+// set event listeners
+
+popupWithImage.setEventListeners();
+
+newCardModal.setEventListeners();
+
 editCardModal.setEventListeners();
 
-const userInfo = new UserInfo({
-  titleSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+deleteCardModal.setEventListeners();
+
+profilePictureModal.setEventListeners();
+
+//button Event Listeners
+
+profileEditButton.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.title;
+  profileDescriptionInput.value = userData.description;
+  editCardModal.open();
+  editFormValidator.enableButton();
 });
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  editCardModal.close();
+profileImageEditButton.addEventListener("click", () => {
+  profilePictureModal.open();
+});
+
+addNewCardButton.addEventListener("click", () => newCardModal.open());
+
+//functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    cardSelector,
+    showPreviewImage,
+    handleLikeCard,
+    handleDeleteCardPopup
+  );
+  section.addItem(card.getView());
+}
+
+function handleAddCardFormSubmit({ title, url }) {
+  addCardModalSubmitButton.textContent = "Saving...";
+  api
+    .postCard({ name: title, link: url })
+    .then((cardData) => {
+      renderCard(cardData, cardListEl);
+      newCardModal.close();
+      addCardFormElement.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addFormValidator.disableButton();
```

Comment:
`.disableButton()` should be called in block `then`, otherwise you'll not be able to click the button if there is a server error